### PR TITLE
Move pull request for domain whitelisting feature.

### PR DESCRIPTION
**Pull request by karlinnolabs**

### CODE REVIEW

**Comment by Cretezy** [comment]
Remove all the `(object)` and I'll accept. :+1:

**Comment by karlinnolabs** [comment]
the (object) is required because you're using @attr decorators which when trying to run this code with Python2 causes the following error "attrs only works with new-style classes".

Using the object subclass doesn't change things in python3 but makes this code work in python2.

**Comment by Cretezy** [comment]
We aren't actively supporting Python 2. Is there a reason you are still using Python 3?

**Comment by karlinnolabs** [comment]
Not all of the libraries in my project are python 3 yet, I'll be spending some time cleaning that up in a few months time but until then I'm stuck on python 2.

**Comment by Cretezy** [comment]
I know @sils wanted to work towards a Python 3-only library, but if it's only a few lines addition, I don't see the harm into supporting both.

**Comment by karlinnolabs** [comment]
thanks, python3 isn't 100% yet, but soon ;)

File: pymessenger2/bot.py [hunk 1]
```diff
@@ -46,6 +46,20 @@ def auth_args(self):
             self._auth_args = auth
         return self._auth_args
 
+    def add_domains_to_whitelist(self, domains):
```
Comment (sils) [inline pymessenger2/bot.py:49]: I'd name it `whitelist_domains` FWIW

File: pymessenger2/bot.py [hunk 2]
```diff
@@ -46,6 +46,20 @@ def auth_args(self):
...
+        }
+
+        request_endpoint = '{0}/me/messenger_profile'.format(self.graph_url)
+        response = requests.post(
+            request_endpoint,
+            params=self.auth_args,
+            json=payload
+        )
+        result = response.json()
+        return result
```
Comment (sils) [inline pymessenger2/bot.py:61]: any reason you can't use `send_raw`?

**Review by sils — CHANGES_REQUESTED** [review]
Ideally I'd squash the commits in the following manner:

- do a `rebase -i` that should get rid of that merge conflict
- squash the first two commits so we don't have `oops` commits in the history
- keep the last one

I'd advise to shorten the commit message shortlogs to make them better skimmable.

Note that the ultimate decision on this is on @Cretezy and I'm only giving an external recommendation as another contributor of the library.

**Comment by Cretezy** [comment]
Merged it. Doesn't hurt to passively support some Python 2.

**Pull request merged**